workspaces/graph-lib/src/endpoints-graph/index.ts

```diff
@@ -310,36 +319,57 @@ export class GraphQueries {
     return this.wrap(node);
   }
 
-  listNodesByType(type: NodeType): NodeListWrapper {
-    return this.wrapList(type, this.index.nodesByType.get(type) || []);
+  listNodesByType(
+    type: NodeType,
+    includeDeleted: boolean = true
+  ): NodeListWrapper {
+    const nodesByType = this.index.nodesByType.get(type) || [];
+    const filteredNodesByType = includeDeleted
+      ? nodesByType
+      : nodesByType.filter((node) => !isNodeDeleted(node));
+    return this.wrapList(type, filteredNodesByType);
   }
 
   //@TODO add singular find* variant
-  listIncomingNeighborsByType(id: NodeId, incomingNeighborType: NodeType) {
+  listIncomingNeighborsByType(
+    id: NodeId,
+    incomingNeighborType: NodeType,
+    includeDeleted: boolean = true
```

Comment:
let's be consistent and use Removed instead of Deleted everywhere

Reply:
Was here to mention the same thing ☺️